compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -1664,7 +1883,10 @@ abstract class GenJSCode extends plugins.PluginComponent
       if (sym.owner == StringClass && !isStringMethodFromObject) {
         genStringCall(tree)
       } else if (isRawJSType(receiver.tpe) && sym.owner != ObjectClass) {
-        genPrimitiveJSCall(tree, isStat)
+        if (!isScalaJSDefinedJSClass(sym.owner) || jsInterop.isExposed(sym))
+          genPrimitiveJSCall(tree, isStat)
```

Comment:
Arg! This is completely wrong! It won't work under separate compilation (if the `sym.owner` class was compiled in a separate run), because `jsInterop.isExposed` only works in the same run. How stupid could I be!?

I'll need to use an annotation instead of a `mutable.Set[Symbol]`.